### PR TITLE
fix: moved non-verification misbehaviour checks to checkForMisbehaviour 

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #2007 

### Commit Message / Changelog Entry

```bash
fix: moved non-verification misbehaviour checks to checkForMisbehaviour 
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/07-tendermint/misbehaviour_handle.go [hunk 1]
```diff
@@ -68,30 +86,6 @@ func (cs ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCode
 // to misbehaviour.Header2
 // Misbehaviour sets frozen height to {0, 1} since it is only used as a boolean value (zero or non-zero).
```
Comment (crodriguezvega) [inline modules/light-clients/07-tendermint/misbehaviour_handle.go:87]: Is there anything we should update in the godoc of this function after this change?

**Review by crodriguezvega — APPROVED** [review]
Hopefully the last 02-client refactor issue! 😅 🚀

**Review by crodriguezvega — APPROVED** [review]

**Review by damiannolan — APPROVED** [review]
Awesome, thanks @charleenfei! 🙏

**Review by AdityaSripal — APPROVED** [review]
Thanks @charleenfei !

**Pull request merged**